iothub/device/src/Authentication/IotHubConnectionInfo.cs

```diff
@@ -97,33 +99,17 @@ public IotHubConnectionInfo(IotHubConnectionStringBuilder builder)
             {
                 SharedAccessSignature = builder.SharedAccessSignature;
             }
+
+            AuthenticationModel = SharedAccessKeyName == null
+                ? AuthenticationModel.SasIndividual
+                : AuthenticationModel.SasGrouped;
         }
 
-        // This constructor is only used for unit testing.
-        internal IotHubConnectionInfo(
-            string ioTHubName = null,
-            string deviceId = null,
-            string moduleId = null,
-            string hostName = null,
-            Uri httpsEndpoint = null,
-            Uri amqpEndpoint = null,
-            string audience = null,
-            string sharedAccessKeyName = null,
-            string sharedAccessKey = null,
-            string sharedAccessSignature = null,
-            bool isUsingGateway = false)
+        // This contructor is intended for creating an IotHubConnectionInfo that uses an IAuthenticationMethod-based authentication.
+        internal IotHubConnectionInfo()
```

Comment:
I've set this constructor skeleton code for us to use when we refactor the implementation of the connection string builder, together with IotHubConnectionInfo.